lib/httpx/plugins/stream_bidi.rb

```diff
@@ -275,6 +275,10 @@ def can_buffer?
         def transition(nextstate)
           return super unless @options.stream
 
+          if nextstate == :idle
```

Comment:
`nextstate` transition management is handled below in the `case` statement. This should be moved there, and mirror what the `when :headers` clause does.